File: Programmers_42583/solution.js
function solution(bridge_length, weight, truck_weights) {
  var answer = 0;
  const bridge = [];
  for (let i = 0; i < bridge_length; i++) {
    bridge.push(0);
  }

  const shift_item = truck_weights.shift();
  bridge[bridge_length - 1] = shift_item;

  let sum = shift_item;

  while (sum > 0) {
    sum -= bridge[0];
    for (let i = 1; i < bridge_length; i++) {
      bridge[i - 1] = bridge[i];
    }
    bridge[bridge_length - 1] = 0;

    const shift_item = truck_weights.shift();
    if (shift_item !== undefined) {
      if (sum + shift_item <= weight) {
        bridge[bridge_length - 1] = shift_item;
        sum += shift_item;
      } else {
        truck_weights.unshift(shift_item);
      }
      answer++;
    } else {
      sum = 0;
    }
  }
  answer += bridge_length + 1;
  return answer;
}

solution(100, 100, [10]);

File: Programmers_42583/solution.py
def solution(bridge_length, weight, truck_weights):
    second = 1
    bridge = [0] * bridge_length
    while 1 :
        # 트럭 하나씩 앞으로 전진
        bridge.pop(0)
        
        # 대기 트럭 없을 때
        if len(truck_weights) == 0 :
            bridge.append(0)
            if sum(bridge) == 0 :
                return second
        #대기 트럭 있을 때
        else :
            pop_truck_weight = truck_weights[0]
            if sum(bridge) + pop_truck_weight <= weight and len(bridge) < bridge_length:
                # 트럭 하나 꺼낸 후 다리에 올리기
                pop_truck = truck_weights.pop(0)
                bridge.append(pop_truck)
            else :
                bridge.append(0)

        second += 1

    return second

solution(	2, 10, [7, 4, 5, 6])